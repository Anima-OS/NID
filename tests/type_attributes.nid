#include <stdio.h>

[[handle,free(fclose)]] FILE* fopen(const char* filename, const char* mode);

 size_t fread([[cast(char*)]] void* ptr, size_t size, size_t nmemb, [[handle]] FILE* stream);

[[destructor]] int fclose(FILE* f);