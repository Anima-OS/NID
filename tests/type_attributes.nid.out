{ declarations: 
   [ { pragma: '#include <stdio.h>' },
     { function: 
        { type: { name: 'FILE', pointer: '*' },
          attributes: { handle: true, free: 'fclose' },
          name: 'fopen',
          parameters: 
           [ { type: { name: 'const char', pointer: '*' },
               attributes: {},
               name: 'filename' },
             { type: { name: 'const char', pointer: '*' },
               attributes: {},
               name: 'mode' } ] } },
     { function: 
        { type: { name: 'size_t' },
          name: 'fread',
          attributes: {},
          parameters: 
           [ { type: { name: 'void', pointer: '*' },
               attributes: { cast: 'char*' },
               name: 'ptr' },
             { type: { name: 'size_t' }, name: 'size', attributes: {} },
             { type: { name: 'size_t' }, name: 'nmemb', attributes: {} },
             { type: { name: 'FILE', pointer: '*' },
               attributes: { handle: true },
               name: 'stream' } ] } },
     { function: 
        { type: { name: 'int' },
          name: 'fclose',
          attributes: { destructor: true },
          parameters: 
           [ { type: { name: 'FILE', pointer: '*' },
               attributes: {},
               name: 'f' } ] } } ] }
